### DISCUSSION TITLE
Get whether a field is valid immediately on change with `mode: onBlur`

### DISCUSSION THREAD

**Discussion by centraldogma99** [react-hook-form/react-hook-form#10262] [Q&A]
Hi, I am trying to make a form made of multiple pages.
A page has one input field and one disabled button.

The form has requirements:

Requirement 1.
**If you enter some value on the input and if it passes validation, the button should be enabled immediately. (not on blur)**
And if you click the button next page with same contents with next field(an input field and a button) shows.
(I used this logic: 'if `formState.errors.[field_name]` is falsy, enable the button.')

Requirement 2.
Also, **input validation error message must show after the input is blurred**.
This is because, if error message appears while entering in input, it is really bad experience.

So my problem is:
- If I use `mode: onBlur`, Req 2 works but Req 1 not works(The button is enabled on blur, not immediately).
- If I use `mode: onChange` or `mode: all`, Req 1 works but Req 2 not works(error message occurs while entering value).

What should I do? I know `formState.isValid` works like this way, but it is about whole form not a field.

**Comment by leapful** [comment]
@centraldogma99 

Just use mode `onBlur` then perform `trigger` inside `onChange` of filed related requirement 1. 

Note: You might need to apply `debounce` for `onChange` to prevent multiple `trigger` calls on end-user typing.
